src/compiler/transformers/ts.ts

```diff
@@ -1688,6 +1689,9 @@ namespace ts {
                 case SyntaxKind.StringKeyword:
                     return createIdentifier("String");
 
+                case SyntaxKind.ObjectKeyword:

```

Comment:
I would move this line down to above 1728, as that set of cases has a `break` to the default behavior of emitting `Object`.